hack/bin/serve-charts.sh

```diff
@@ -2,16 +2,18 @@
 
 set -euo pipefail
 
-: ${HELM_SERVER_PORT:=4001}
+: "${HELM_SERVER_PORT:=4001}"
 
 # get rid of all helm repositories
 # We need to deal with helm repo list failing because of https://github.com/helm/helm/issues/10028
 (helm repo list -o json || echo '[]') | jq -r '.[] | .name' | xargs -I% helm repo remove %
 
-cd "$(dirname "$BASH_SOURCE[0]")/../../.local/charts"
-for chart in $@; do
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+cd "$SCRIPT_DIR/../../.local/charts"
```

Comment:
**thought:** This seems a bit convoluted to me: we temporarily change directory just to print its name, then cd into that again.

**suggestion:** I think using [`dirname`](https://explainshell.com/explain/1/dirname) and [`realpath`](https://explainshell.com/explain/1/realpath) should do the same thing, are easier to understand and thus common practice:

```suggestion
SCRIPT_DIR=$(realpath "$(dirname "${BASH_SOURCE[0]}")")
cd "$SCRIPT_DIR/../../.local/charts"
```

Reply:
the solution is from https://stackoverflow.com/questions/59895/how-do-i-get-the-directory-where-a-bash-script-is-located-from-within-the-script It probably went through a lot of discussion, so I wouldn't wanna change it